### ISSUE TITLE
TreeView component needs to be in sync with current design

### ISSUE THREAD

**Issue by jianliao** [adobe/spectrum-css#618]
## Description
TreeView design has graduated from contribution to beta. The current spectrum-css implementation needs to change to match it. Please check this slack thread for more detail: https://adobedesign.slack.com/archives/GEFETJ9BR/p1583183591233400

## Steps to reproduce

1. Go to https://opensource.adobe.com/spectrum-css/components/treeview/
2. Observe and compare with XD file

## Expected behavior
The implementation should match design

## Screenshots
Current impl:
![image](https://user-images.githubusercontent.com/1207520/75939521-4ca42d80-5e3f-11ea-9f59-e4be4a6d161b.png)
Design: 
![image (1)](https://user-images.githubusercontent.com/1207520/75939557-5e85d080-5e3f-11ea-8ca5-e9279f281a5d.png)

## Environment
Chrome/Firefox/Safari on latest MacOS and IE11/Edge on Windows 10

**Comment by lazd** [comment]
Looks like the height is not a problem, but I do see some alignment issues, which I fixed.

There are some differences in hover behavior that would necessitate some serious markup changes, so I'll have to get with the pixel people and figure out if that's what we're doing.
